relay-general/src/processor/attrs.rs

```diff
@@ -212,6 +213,61 @@ pub struct FieldAttrs {
     pub retain: bool,
 }
 
+/// A set of characters allowed or denied for a (string) field.
+///
+/// Note that this field is generated in the derive, it can't be constructed easily in tests.
+#[derive(Clone, Copy)]
+pub struct CharacterSet {
+    /// Generated in derive for performance. Can be left out when set is created manually.
+    pub is_match: Option<fn(char) -> bool>,
```

Comment:
_nit_: What if you always assign a default function that returns `true` or `false`? I would not even allow creating this struct manually, ever.

Reply:
answering this and the other questions: I need ranges and is_negative purely for jsonschema. `is_match` field alone is sufficient for basic functionality.